main.cpp

```diff
@@ -7,12 +7,32 @@
 #include "toz3Options.h"
 
 
+class Pruner : public Transform{
+    public: 
+        Pruner(){
+            setName("Pruner");
+        }
+
+        IR::Node* preorder(IR::Statement *s);
+
+};
+
+
+IR::Node* Pruner::preorder(IR::Statement *s){
```

Comment:
This should be IR::AssignmentStatement, then you do not have to do that manual check in line 22 below.